File: mobile/android/app/src/main/java/com/beijixing/app/data/model/CrawlModels.kt
package com.beijixing.app.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime

@Parcelize
data class CrawlTask(
    val id: Long? = null,
    val taskName: String? = null,
    val taskType: String? = null,
    val platformCode: String? = null,
    val targetId: String? = null,
    val keywords: String? = null,
    val maxCrawlCount: Int? = null,
    val crawlIntervalSeconds: Int? = null,
    val totalCommentsFound: Int? = null,
    val highIntentCount: Int? = null,
    val leadsGenerated: Int? = null,
    val messagesSent: Int? = null,
    val status: Int? = null,
    val progressPercent: Int? = null,
    val errorMessage: String? = null,
    val startTime: LocalDateTime? = null,
    val endTime: LocalDateTime? = null,
    val createTime: LocalDateTime? = null,
    val updateTime: LocalDateTime? = null
) : Parcelable

@Parcelize
data class SocialComment(
    val id: Long? = null,
    val commentId: String? = null,
    val crawlTaskId: Long? = null,
    val platformCode: String? = null,
    val contentId: String? = null,
    val authorId: String? = null,
    val authorName: String? = null,
    val authorAvatar: String? = null,
    val authorBio: String? = null,
    val userFollowerCount: Long? = null,
    val userFollowingCount: Long? = null,
    val userVerified: Boolean? = null,
    val commentText: String? = null,
    val likeCount: Int? = null,
    val replyCount: Int? = null,
    val publishTime: LocalDateTime? = null,
    val extractedPhone: String? = null,
    val extractedWechat: String? = null,
    val hasPhoneContact: Boolean? = false,
    val hasWechatContact: Boolean? = false,
    val aiIntentScore: Int? = null,
    val aiIntentLevel: String? = null,
    val isHighIntent: Boolean? = false,
    val aiAnalysisResult: String? = null,
    val leadGenerated: Boolean? = false,
    val generatedLeadId: Long? = null,
    val messageSent: Boolean? = false,
    val messageSentTime: LocalDateTime? = null,
    val deleted: Int? = 0,
    val createTime: LocalDateTime? = null,
    val updateTime: LocalDateTime? = null
) : Parcelable

@Parcelize
data class MessageTemplate(
    val id: Long? = null,
    val templateName: String? = null,
    val platformCode: String? = null,
    val intentLevel: String? = null,
    val templateContent: String? = null,
    val templateVariables: String? = null,
    val aiGenerated: Boolean? = false,
    val isDefault: Boolean? = false,
    val useCount: Int? = null,
    val successRate: Double? = null,
    val replyRate: Double? = null,
    val status: Int? = 1,
    val createTime: LocalDateTime? = null,
    val updateTime: LocalDateTime? = null
) : Parcelable

@Parcelize
data class MessageResult(
    val success: Boolean = false,
    val commentId: Long? = null,
    val platformCode: String? = null,
    val authorId: String? = null,
    val authorName: String? = null,
    val messageContent: String? = null,
    val templateId: Long? = null,
    val templateName: String? = null,
    val errorMessage: String? = null,
    val sendTime: LocalDateTime? = null
) : Parcelable

@Parcelize
data class LeadResult(
    val success: Boolean = false,
    val generatedCount: Int = 0,
    val skippedCount: Int = 0,
    val errors: Map<Long, String>? = null,
    val generatedLeads: List<CrawlLead>? = null
) : Parcelable

@Parcelize
data class CrawlLead(
    val id: Long? = null,
    val title: String? = null,
    val source: String? = null,
    val channel: String? = null,
    val customerName: String? = null,
    val customerPhone: String? = null,
    val level: String? = null,
    val intentScore: Int? = null,
    val requirementDesc: String? = null,
    val status: String? = null,
    val createTime: LocalDateTime? = null
) : Parcelable

data class SendMessageRequest(
    val commentId: Long,
    val templateId: Long?,
    val content: String
)

data class BatchSendMessageRequest(
    val commentIds: List<Long>,
    val templateId: Long?,
    val maxConcurrent: Int = 5,
    val intervalMs: Int = 30000
)

data class GenerateLeadsRequest(
    val minScore: Int? = null,
    val requiredLevels: List<String>? = null,
    val requireContactInfo: Boolean = false,
    val minFollowerCount: Int? = null,
    val autoAssign: Boolean = true,
    val generateFollowUpTask: Boolean = true
)

data class BatchGenerateLeadsRequest(
    val commentIds: List<Long>,
    val autoAssign: Boolean = true,
    val generateFollowUpTask: Boolean = true
)

File: mobile/ios/BeijiXingApp/Models/CrawlModels.swift
import Foundation

struct CrawlTask: Codable, Identifiable {
    let id: Int64?
    var taskName: String?
    var taskType: String?
    var platformCode: String?
    var targetId: String?
    var keywords: String?
    var maxCrawlCount: Int?
    var crawlIntervalSeconds: Int?
    var totalCommentsFound: Int?
    var highIntentCount: Int?
    var leadsGenerated: Int?
    var messagesSent: Int?
    var status: Int?
    var progressPercent: Int?
    var errorMessage: String?
    var startTime: String?
    var endTime: String?
    var createTime: String?
    var updateTime: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case taskName = "task_name"
        case taskType = "task_type"
        case platformCode = "platform_code"
        case targetId = "target_id"
        case keywords
        case maxCrawlCount = "max_crawl_count"
        case crawlIntervalSeconds = "crawl_interval_seconds"
        case totalCommentsFound = "total_comments_found"
        case highIntentCount = "high_intent_count"
        case leadsGenerated = "leads_generated"
        case messagesSent = "messages_sent"
        case status
        case progressPercent = "progress_percent"
        case errorMessage = "error_message"
        case startTime = "start_time"
        case endTime = "end_time"
        case createTime = "create_time"
        case updateTime = "update_time"
    }
    
    var platformDisplayName: String {
        switch platformCode?.uppercased() {
        case "DOUYIN":
            return "抖音"
        case "XIAOHONGSHU":
            return "小红书"
        case "KUAISHOU":
            return "快手"
        case "WEIBO":
            return "微博"
        case "BILIBILI":
            return "B站"
        default:
            return platformCode ?? "未知"
        }
    }
    
    var statusText: String {
        switch status {
        case 0:
            return "待执行"
        case 1:
            return "进行中"
        case 2:
            return "已完成"
        case 3:
            return "失败"
        default:
            return "未知"
        }
    }
}

struct SocialComment: Codable, Identifiable {
    let id: Int64?
    var commentId: String?
    var crawlTaskId: Int64?
    var platformCode: String?
    var contentId: String?
    var authorId: String?
    var authorName: String?
    var authorAvatar: String?
    var authorBio: String?
    var userFollowerCount: Int64?
    var userFollowingCount: Int64?
    var userVerified: Bool?
    var commentText: String?
    var likeCount: Int?
    var replyCount: Int?
    var publishTime: String?
    var extractedPhone: String?
    var extractedWechat: String?
    var hasPhoneContact: Bool?
    var hasWechatContact: Bool?
    var aiIntentScore: Int?
    var aiIntentLevel: String?
    var isHighIntent: Bool?
    var aiAnalysisResult: String?
    var leadGenerated: Bool?
    var generatedLeadId: Int64?
    var messageSent: Bool?
    var messageSentTime: String?
    var deleted: Int?
    var createTime: String?
    var updateTime: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case commentId = "comment_id"
        case crawlTaskId = "crawl_task_id"
        case platformCode = "platform_code"
        case contentId = "content_id"
        case authorId = "author_id"
        case authorName = "author_name"
        case authorAvatar = "author_avatar"
        case authorBio = "author_bio"
        case userFollowerCount = "user_follower_count"
        case userFollowingCount = "user_following_count"
        case userVerified = "user_verified"
        case commentText = "comment_text"
        case likeCount = "like_count"
        case replyCount = "reply_count"
        case publishTime = "publish_time"
        case extractedPhone = "extracted_phone"
        case extractedWechat = "extracted_wechat"
        case hasPhoneContact = "has_phone_contact"
        case hasWechatContact = "has_wechat_contact"
        case aiIntentScore = "ai_intent_score"
        case aiIntentLevel = "ai_intent_level"
        case isHighIntent = "is_high_intent"
        case aiAnalysisResult = "ai_analysis_result"
        case leadGenerated = "lead_generated"
        case generatedLeadId = "generated_lead_id"
        case messageSent = "message_sent"
        case messageSentTime = "message_sent_time"
        case deleted
        case createTime = "create_time"
        case updateTime = "update_time"
    }
    
    var displayIntentLevel: String {
        switch aiIntentLevel?.uppercased() {
        case "A":
            return "A级-超高意向"
        case "B":
            return "B级-高意向"
        case "C":
            return "C级-中意向"
        case "D":
            return "D级-低意向"
        default:
            return aiIntentLevel ?? "未分析"
        }
    }
}

struct MessageTemplate: Codable, Identifiable {
    let id: Int64?
    var templateName: String?
    var platformCode: String?
    var intentLevel: String?
    var templateContent: String?
    var templateVariables: String?
    var aiGenerated: Bool?
    var isDefault: Bool?
    var useCount: Int?
    var successRate: Double?
    var replyRate: Double?
    var status: Int?
    var createTime: String?
    var updateTime: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case templateName = "template_name"
        case platformCode = "platform_code"
        case intentLevel = "intent_level"
        case templateContent = "template_content"
        case templateVariables = "template_variables"
        case aiGenerated = "ai_generated"
        case isDefault = "is_default"
        case useCount = "use_count"
        case successRate = "success_rate"
        case replyRate = "reply_rate"
        case status
        case createTime = "create_time"
        case updateTime = "update_time"
    }
}

struct MessageResult: Codable {
    let success: Bool
    let commentId: Int64?
    let platformCode: String?
    let authorId: String?
    let authorName: String?
    let messageContent: String?
    let templateId: Int64?
    let templateName: String?
    let errorMessage: String?
    let sendTime: String?
    
    enum CodingKeys: String, CodingKey {
        case success
        case commentId = "comment_id"
        case platformCode = "platform_code"
        case authorId = "author_id"
        case authorName = "author_name"
        case messageContent = "message_content"
        case templateId = "template_id"
        case templateName = "template_name"
        case errorMessage = "error_message"
        case sendTime = "send_time"
    }
}

struct LeadResult: Codable {
    let success: Bool
    let generatedCount: Int?
    let skippedCount: Int?
    let errors: [String: String]?
    let generatedLeads: [Lead]?
}

struct Lead: Codable, Identifiable {
    let id: Int64?
    var title: String?
    var source: String?
    var channel: String?
    var customerName: String?
    var customerPhone: String?
    var level: String?
    var intentScore: Int?
    var requirementDesc: String?
    var status: String?
    var createTime: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case title
        case source
        case channel
        case customerName = "customer_name"
        case customerPhone = "customer_phone"
        case level
        case intentScore = "intent_score"
        case requirementDesc = "requirement_desc"
        case status
        case createTime = "create_time"
    }
}
